test/ExpressiveInstallerTest/RoutersTest.php

```diff
@@ -93,9 +106,27 @@ public function testRouter(
         );
 
         // Test home page
-        $setupRoutes = (strpos($copyFilesKey, 'minimal') !== 0);
+        $setupRoutes = strpos($copyFilesKey, 'minimal') !== 0;
         $response = $this->getAppResponse('/', $setupRoutes);
         $this->assertEquals($expectedResponseStatusCode, $response->getStatusCode());
+
+        /** @var Application $app */
+        $app = $container->get(Application::class);
+        $this->assertCount(count($expectedRoutes), $app->getRoutes());
+        foreach ($app->getRoutes() as $route) {
+            foreach ($expectedRoutes as $expectedRoute) {
+                if ($expectedRoute['name'] === $route->getName()) {
+                    $this->assertEquals($expectedRoute['path'], $route->getPath());
+                    // todo: not sure how we can test middleware?
+                    // $this->assertEquals($expectedRoute['middleware'], $route->getMiddleware());
```

Comment:
We need to do the following:

- Split the logic currently in `ProjectSandboxTrait` into two separate methods:
  - one for setting up the application, pipeline, and routes
  - one for dispatching the provided `$path`
- Update all calls in the test suite to `getAppResponse()` to instead call the above two methods in succession.
- In this particular method, we'd call the method for setup first, and then loop over the routes, passing the path for each to `getAppResponse()`. 
  - For the "minimal" cases, we'd need to have a way of providing the routes *without injecting them* so we can explicitly test the 404 cases.